src/tools/rbd/Utils.cc

```diff
@@ -367,53 +369,27 @@ int get_image_options(const boost::program_options::variables_map &vm,
     object_size = vm[at::IMAGE_OBJECT_SIZE].as<uint64_t>();
     order = std::round(std::log2(object_size)); 
   } else {
-    order = g_conf->rbd_default_order;
+    order_specified = false;
   }
 
   if (vm.count(at::IMAGE_FEATURES)) {
     features = vm[at::IMAGE_FEATURES].as<uint64_t>();
     features_specified = true;
-  } else {
-    features = g_conf->rbd_default_features;
   }
 
   if (vm.count(at::IMAGE_STRIPE_UNIT)) {
     stripe_unit = vm[at::IMAGE_STRIPE_UNIT].as<uint64_t>();
-  } else {
-    stripe_unit = g_conf->rbd_default_stripe_unit;
+    stripe_specified = true;
   }
 
   if (vm.count(at::IMAGE_STRIPE_COUNT)) {
     stripe_count = vm[at::IMAGE_STRIPE_COUNT].as<uint64_t>();
-  } else {
-    stripe_count = g_conf->rbd_default_stripe_count;
-  }
-
-  if ((stripe_unit != 0 && stripe_count == 0) ||
-      (stripe_unit == 0 && stripe_count != 0)) {
-    std::cerr << "must specify both (or neither) of stripe-unit and stripe-count"
-              << std::endl;
-    return -EINVAL;
-  } else if (stripe_unit || stripe_count) {
-    if ((1ull << order) % stripe_unit || stripe_unit >= (1ull << order)) {
-      std::cerr << "stripe unit is not a factor of the object size" << std::endl;
-      return -EINVAL;
-    }
-    if (stripe_count == 1) {
-      std::cerr << "stripe count not allowed to be 1" << std::endl;
-      return -EINVAL;
-    }
-    features |= RBD_FEATURE_STRIPINGV2;
-  } else {
-    if (features_specified && ((features & RBD_FEATURE_STRIPINGV2) != 0)) {
-      std::cerr << "must specify both of stripe-unit and stripe-count when specify striping features" << std::endl;
-      return -EINVAL;
-    }
-    features &= ~RBD_FEATURE_STRIPINGV2;
+    stripe_specified = true;
   }
 
   if (vm.count(at::IMAGE_SHARED) && vm[at::IMAGE_SHARED].as<bool>()) {
     features &= ~RBD_FEATURES_SINGLE_CLIENT;
```

Comment:
The ```features``` haven't potentially been initialized to the default feature set at this time if ```!features_specified```